core/src/test/scala/org/apache/spark/AccumulatorSuite.scala

```diff
@@ -237,6 +236,65 @@ class AccumulatorSuite extends SparkFunSuite with Matchers with LocalSparkContex
     acc.merge("kindness")
     assert(acc.value === "kindness")
   }
+
+  test("updating garbage collected accumulators") {
```

Comment:
what does this test do? prove accumulator can be GCed even it's valid? The map in  `AccumulatorContext` is fixed-size so this definitely can happen and we don't need to prove it.

Reply:
It demonstrates that it's possible to receive an update for a GCed accumulator which crashes the app.

Reply:
This test can reproduce the crush scenario in original code base and successful ended after this patch. I think @cloud-fan is worrying about this test shouldn't commit in code base because it complexity？

Reply:
tests also have a maintenance cost. Sometimes we change something and break the test, we need to look into the test and see if the test is wrong or the change is wrong. Or sometimes the test becomes flaky and we need to investigate.

This test seems to prove a thing can happen while it's already proved by other tests, and I think this test is not worth the maintenance cost given its complexity.

Reply:
I agree this test is quite ugly. Let me just revert the last commit.